Calendar/BookingHelpers/BookingMBHelpers.ps1

```diff
@@ -0,0 +1,238 @@
+﻿function Get-BookingMBData {
+    param (
+        [string]$Identity
+    )
+
+    $script:BMB = Get-Mailbox -Identity $Identity -ErrorAction SilentlyContinue
+    if ($null -eq $script:BMB) {
+        Write-DashLineBoxColor -Line "Booking Mailbox not found" -Color "Red"
+        return $null
+    }
+
+    $bookingMBData = [PSCustomObject]@{
+        DisplayName                   = $script:BMB.DisplayName
+        Identity                      = $script:BMB.Identity
+        RecipientType                 = $script:BMB.RecipientType  #: UserMailbox
+        RecipientTypeDetails          = $script:BMB.RecipientTypeDetails  # SchedulingMailbox
+        EmailAddresses                = $script:BMB.EmailAddresses
+        IsMailboxEnabled              = $script:BMB.IsMailboxEnabled
+        HiddenFromAddressListsEnabled = $script:BMB.HiddenFromAddressListsEnabled
+        IsSoftDeletedByRemove         = $script:BMB.IsSoftDeletedByRemove
+        IsSoftDeletedByDisable        = $script:BMB.IsSoftDeletedByDisable
+        IsInactiveMailbox             = $script:BMB.IsInactiveMailbox
+        WhenSoftDeleted               = $script:BMB.WhenSoftDeleted
+        WindowsEmailAddress           = $script:BMB.WindowsEmailAddress
+        WhenCreated                   = $script:BMB.WhenCreated
+        Guid                          = $script:BMB.Guid
+        OriginatingServer             = $script:BMB.OriginatingServer
+    }
+    return $bookingMBData
+}
+
+
+
+
+function Get-GraphBusiness {
+    param (
+        [string]$Identity
+    )
+
+    $MB = Get-MgBookingBusiness -BookingBusinessId $Identity
+
+    Write-Host "Exporting Working Hours"
+
+    $a=$null
+    $a=@()
+    $a+= [PSCustomObject]@{
+        DisplayName         = $MB.DisplayName
+        OwnerEmail          = $MB.Email
+        IsPublished         = $MB.IsPublished
+        DefaultCurrencyIso  = $MB.DefaultCurrencyIso
+        DefaultTimeZone     = $MB.DefaultTimeZone
+        LanguageTag         = $MB.LanguageTag
+        Phone               = $MB.Phone
+        PublicUrl           = $MB.PublicUrl
+        WebSiteUrl          = $MB.WebSiteUrl
+        CreatedDateTime     = $MB.AdditionalProperties["createdDateTime"]
+        lastUpdatedDateTime = $MB.AdditionalProperties["lastUpdatedDateTime"]
+        Address             = "City= " + $MB.Address.City + ", `r`n" +
+        "CountryOrRegion= " + $MB.Address.CountryOrRegion + ", `r`n" +
+        "PostalCode= " + $MB.Address.PostalCode + ", `r`n" +
+        "State= " + $MB.Address.State + ", `r`n" +
+        "Street= " + $MB.Address.Street
+    }
+
+    return $a
```

Comment:
>$a [](http://example.com/codeflow?start=11&length=2)

Don't be lazy.

Reply:
Here and else where.

Reply:
not being lazy, but this comment is being sent from VSCode